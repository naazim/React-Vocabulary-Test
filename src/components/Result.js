import React from 'react';
import ResultItem from './ResultItem';

const Result = (props) => {
    const filtered = props.shuffledList.filter((word) => word.foreign === word.userAnswer);
    const score = filtered.length / 20 * 100;

    /**
     * @description Returns the user to the Create Quiz list page
     */
    const handleSubmit = () => {
        props.handleModule('CreateQuiz');
    };

    return (
        <div className="wrapper">
            <div className="result__header">
                <h1 className="result__title">Your score</h1>
                <div className="result__score">{score}%</div>
            </div>

            {props.shuffledList.map((word, index) => (
                <ResultItem key={index} word={word} index={index} />
            ))}

            <button
                onClick={handleSubmit}
                className="question__button question__button-next"
            >ðŸ¡¸</button>
        </div>
    );
};

export default Result;