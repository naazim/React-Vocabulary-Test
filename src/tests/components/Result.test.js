import React from 'react';
import { shallow } from 'enzyme';
import Result from '../../components/Result';

describe('Result', () => {
    const handleModuleSpy = jest.fn();
    const shuffledList=[{"native":"Hello","foreign":"Hallo"},{"native":"Good","foreign":"Gut"}];
    const wrapper = shallow(<Result shuffledList={shuffledList} handleModule={handleModuleSpy} />);

    test('should render Result correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('should render a result item for every object in the shuffled list', () => {
        expect(wrapper.find('ResultItem').length).toBe(2);
    });

    test('should switch to the list component on clicking back button', () => {
        wrapper.find('.question__button').simulate('click');
        expect(handleModuleSpy).toHaveBeenCalled();
    })
});