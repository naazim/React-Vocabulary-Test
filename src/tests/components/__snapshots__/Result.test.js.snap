// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result should render Result correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="result__header"
  >
    <h1
      className="result__title"
    >
      Your score
    </h1>
    <div
      className="result__score"
    >
      0
      %
    </div>
  </div>
  <ResultItem
    index={0}
    key="0"
    word={
      Object {
        "foreign": "Hallo",
        "native": "Hello",
      }
    }
  />
  <ResultItem
    index={1}
    key="1"
    word={
      Object {
        "foreign": "Gut",
        "native": "Good",
      }
    }
  />
  <button
    className="question__button question__button-next"
    onClick={[Function]}
  >
    ðŸ¡¸
  </button>
</div>
`;
